packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -99,9 +107,10 @@ function createIframeVirtualEnvironment(
         blueCreateHooksCallbackCache.set(blueRefs.document, blueConnector);
     }
     const { eval: redIndirectEval } = redWindow;
+    const signedConnectorEval = (v: string) => redIndirectEval(trustedCreateScript(v));
```

Comment:
This poses a very good question actually. Should we always call the callback? not only for connector, but for everything that gets evaluated via the membrane? The thing is that in order for you to evaluate here, you MUST be privileged. We could say that if you gain access to evaluateInSandbox, then you're off the hook, because it signs for you. Food for thoughts.

Reply:
Currently evaluateInSandbox does sign before calling `eval` in the Virtual Environment. We could try it out as it is and if we want to move the signing process to a lower level (near-membrane) we could very well do that since it's a trivial change.

Reply:
The thing about reusing this callback for top/location is that this is used for our Marshall eval which we don't want to transform with top/location.

Reply:
but we could! nothing really prevent us from today that as well. certainly it will not affect the marshaling code.